### PR TITLE
Fix and cleanup dataset methods; add tests

### PR DESCRIPTION
**Pull request by zaucker**
- Consistent return values for dataset methods 
- Add tests
- Switch to version 6.0.0 rc1

### CODE REVIEW

File: lib/Agrammon/DB/Dataset.pm6 [hunk 1]
```diff
@@ -266,6 +266,7 @@ class Agrammon::DB::Dataset does Agrammon::DB {
              ORDER BY data_instance_order ASC, data_var
             DATASET
             $!data = $results.arrays;
+            $!comment
         }
         return self;
```
Comment (jnthn) [inline lib/Agrammon/DB/Dataset.pm6:271]: This `$!comment` seems useless given we `return self` immediately below, so it cannot be the return value.

File: lib/Agrammon/DB/Dataset.pm6 [hunk 2]
```diff
@@ -329,21 +329,21 @@ class Agrammon::DB::Dataset does Agrammon::DB {
...
 
             $ret = $db.query(q:to/SQL/, $comment, $username, $!name, $variable);
                 INSERT INTO data_new (data_dataset, data_var, data_comment)
                      VALUES          (dataset_name2id($2,$3), $4, $1)
                 RETURNING data_comment
                 SQL
+            return $ret.rows;
 
             # couldn't save comment
             die X::Agrammon::DB::Dataset::StoreInputCommentFailed.new(:$comment, :$variable) unless $ret.rows;
```
Comment (jnthn) [inline lib/Agrammon/DB/Dataset.pm6:342]: This `die` is unreachable because it's after a `return`.

File: lib/Agrammon/DB/Dataset.pm6 [hunk 3]
```diff
@@ -354,30 +354,32 @@ class Agrammon::DB::Dataset does Agrammon::DB {
...
 
             $ret = $db.query(q:to/SQL/, $comment, $username, $!name, $variable, $instance);
                 INSERT INTO data_new (data_dataset, data_var, data_comment, data_instance)
                      VALUES          (dataset_name2id($2,$3), $4, $1, $5)
                 RETURNING data_comment
             SQL
 
+            return $ret.rows;
             # couldn't save comment
             die X::Agrammon::DB::Dataset::StoreInputCommentFailed.new(:$comment, :$variable) unless $ret.rows;
```
Comment (jnthn) [inline lib/Agrammon/DB/Dataset.pm6:367]: Another unreachable `die`

File: lib/Agrammon/DB/Dataset.pm6 [hunk 4]
```diff
@@ -354,30 +354,32 @@ class Agrammon::DB::Dataset does Agrammon::DB {
...
         }
     }
 
-    method store-input-comment(:$variable!, :$comment --> Nil) {
+    method store-input-comment($variable!, $comment) {
```
Comment (jnthn) [inline lib/Agrammon/DB/Dataset.pm6:371]: Positional parameters are required by default

File: lib/Agrammon/DB/Dataset.pm6 [hunk 5]
```diff
@@ -387,20 +389,20 @@ class Agrammon::DB::Dataset does Agrammon::DB {
...
+            return $ret.rows if $ret.rows;
 
             $ret = $db.query(q:to/SQL/, $value, $username, $!name, $variable);
                 INSERT INTO data_new (data_dataset, data_var, data_val)
                      VALUES          (dataset_name2id($2,$3), $4, $1)
                 RETURNING data_val
             SQL
+            return $ret.rows;
 
             die X::Agrammon::DB::Dataset::StoreDataFailed.new($variable) unless $ret.rows;
```
Comment (jnthn) [inline lib/Agrammon/DB/Dataset.pm6:401]: Dead `die`

File: lib/Agrammon/DB/Dataset.pm6 [hunk 6]
```diff
@@ -411,19 +413,20 @@ class Agrammon::DB::Dataset does Agrammon::DB {
...
+            return $ret.rows if $ret.rows;
 
             $ret = $db.query(q:to/SQL/, $value, $username, $!name, $variable, $instance);
                 INSERT INTO data_new (data_dataset, data_var, data_val, data_instance)
                      VALUES (dataset_name2id($2,$3), $4, $1, $5)
                 RETURNING data_comment
             SQL
+            return $ret.rows;
 
             die X::Agrammon::DB::Dataset::StoreDataFailed.new($variable) unless $ret.rows;
```
Comment (jnthn) [inline lib/Agrammon/DB/Dataset.pm6:425]: Another one

File: lib/Agrammon/Web/Service.pm6 [hunk 7]
```diff
@@ -273,8 +273,8 @@ class Agrammon::Web::Service {
         }
     }
 
-    method store-input-comment(Agrammon::Web::SessionUser $user, :$dataset!, :$variable!, :$comment --> Nil) {
-        Agrammon::DB::Dataset.new(:$user, :name($dataset)).store-input-comment(:$variable, :$comment);
+    method store-input-comment(Agrammon::Web::SessionUser $user, $dataset!, $variable!, $comment --> Nil) {
```
Comment (jnthn) [inline lib/Agrammon/Web/Service.pm6:276]: Positional parameters are required by default

File: t/dataset.t [hunk 8]
```diff
@@ -0,0 +1,196 @@
...
+    is $dataset.comment,   'Test comment',     'User has correct comment';
+    is $dataset.version,     '6.0-test',     'User has correct version';
+    is $dataset.records,     42,     'User has correct records';
+    is $dataset.mod-date,    $date,     'User has correct mod-date';
+    is $dataset.data.elems,  0,     'User has 0 data';
+    is $dataset.tags.elems,  0,     'User has 0 tags';
+#    is $dataset.user.username,   'agtestuser',     'User has correct user';
+}
+
+# done-testing; exit;
```
Comment (jnthn) [inline t/dataset.t:69]: Leftover things to clean up?

File: lib/Agrammon/DB/Dataset.pm6 (hunk 1 above)
Comment (zaucker) [reply lib/Agrammon/DB/Dataset.pm6:271]: Indeed, that's a left over from I don't know what ... removed.

File: lib/Agrammon/DB/Dataset.pm6 (hunk 2 above)
Comment (zaucker) [reply lib/Agrammon/DB/Dataset.pm6:342]: Added the `return` at the wrong place, glad you caught it. Fixed.

File: lib/Agrammon/DB/Dataset.pm6 (hunk 3 above)
Comment (zaucker) [reply lib/Agrammon/DB/Dataset.pm6:367]: Fixed.

File: lib/Agrammon/DB/Dataset.pm6 (hunk 4 above)
Comment (zaucker) [reply lib/Agrammon/DB/Dataset.pm6:371]: As Comma told, but I didn't pay attention ... fixed.

File: lib/Agrammon/DB/Dataset.pm6 (hunk 5 above)
Comment (zaucker) [reply lib/Agrammon/DB/Dataset.pm6:401]: Fixed

File: lib/Agrammon/DB/Dataset.pm6 (hunk 6 above)
Comment (zaucker) [reply lib/Agrammon/DB/Dataset.pm6:425]: Fixed

File: lib/Agrammon/Web/Service.pm6 (hunk 7 above)
Comment (zaucker) [reply lib/Agrammon/Web/Service.pm6:276]: Fixed. Those were none-positional arguments before and again I didn't pay attention to Comma.

File: t/dataset.t (hunk 8 above)
Comment (zaucker) [reply t/dataset.t]: And a bad one ... prevented the tests from running ... at least I had a plan ... I just wonder way GHA didn't fail.

File: lib/Agrammon/DB/Dataset.pm6 [hunk 9]
```diff
@@ -65,7 +65,7 @@ class X::Agrammon::DB::Dataset::InstanceRenameFailed is Exception {
     }
 }
 
-#| Error when instance couldn't be renamed.
+#| Error when input comment couldn't be renamed.
```
Comment (jnthn) [inline lib/Agrammon/DB/Dataset.pm6:68]: Is "renamed" really accurate here? It's "save" in the message and "store" in the class name.

File: lib/Agrammon/DB/Dataset.pm6 [hunk 10]
```diff
@@ -74,6 +74,15 @@ class X::Agrammon::DB::Dataset::StoreInputCommentFailed is Exception {
     }
 }
 
+#| Error when dataset comment couldn't be renamed.
```
Comment (jnthn) [inline lib/Agrammon/DB/Dataset.pm6:77]: Similar comment comment here.

Comment (zaucker) [reply lib/Agrammon/DB/Dataset.pm6:77]: Fixed

File: lib/Agrammon/DB/Dataset.pm6 (hunk 9 above)
Comment (zaucker) [reply lib/Agrammon/DB/Dataset.pm6:68]: Fixed

**Review by jnthn — APPROVED** [review]

**Pull request merged**